File: 5-longest-palindromic-substring/longest-palindromic-substring.cs
public class Solution {
    public string LongestPalindrome(string s) {
        if (string.IsNullOrEmpty(s)) return "";

        int start = 0, maxLength = 0;
        for(int i = 0; i < s.Length; i++)
        {
            ExpandAroundCenter(s, i, i, ref start, ref maxLength);

            ExpandAroundCenter(s, i, i + 1, ref start, ref maxLength);
        }

        return s.Substring(start,maxLength);

    }

        private void ExpandAroundCenter(string s, int left, int right, ref int  start, ref int maxLength) {
        while (left >= 0 && right < s.Length && s[left] == s[right]) {
            int length = right - left + 1;
            if (length > maxLength) {
                start = left;
                maxLength = length;
            }
            left--;
            right++;
        }
    }
}

File: 5-longest-palindromic-substring/longest-palindromic-substring.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        if(s == "") return "";
        int start = 0, len = 0;
        for(int i = 0; i < s.size(); i++)
        {
            int l = i, r = i;
            while(l >= 0 && r < s.size() && s[l] == s[r])
            {
                if((r - l + 1) > len)
                {
                    start = l;
                    len = r - l + 1;
                }
                l -= 1;
                r += 1;
            }

            l = i, r = i + 1;
            while(l >= 0 && r < s.size() && s[l] == s[r])
            {
                if((r - l + 1) > len)
                {
                    start = l;
                    len = r - l + 1;
                }
                l -= 1;
                r += 1;
            }
        }

        return s.substr(start, len);
    }
};